src/tropical_gemms.jl

```diff
@@ -0,0 +1,44 @@
+# Algebra for the following types
+# the general API will be given as matmul!(A, B, C)
+
+function dims_match(A::T1, B::T2, C::T3) where{T1, T2, T3}
+    MA, KA = size(A)
+    KB, NB = size(B)
+    MC, NC = size(C)
+
+    @assert MA == MC
+    @assert NB == NC
+    @assert KA == KB
+
+    return MA, NB, KA
+end
+
+for (TA, CTA, tA) in [(:CuMatrix, :CuMatrix, 'N'), (:CTranspose, :(Transpose{<:Any, <:StridedCuVecOrMat}), 'T')]
```

Comment:
CTA and CTB are not used.

Reply:
When I try using the new GEMM in GenericTensorNetworks, I find the following error:
```julia
cuda patch: Error During Test at /home/leo/.julia/dev/GenericTensorNetworks/test/cuda.jl:8
  Got exception outside of a @test
  MethodError: no method matching matmul!(::LinearAlgebra.Transpose{CountingTropicalF64, CuArray{CountingTropicalF64, 2, CUDA.Mem.DeviceBuffer}}, ::LinearAlgebra.Transpose{CountingTropicalF64, CuArray{CountingTropicalF64, 2, CUDA.Mem.DeviceBuffer}}, ::CuArray{CountingTropicalF64, 2, CUDA.Mem.DeviceBuffer})
```

This is probably due to the incorrect dispatch.